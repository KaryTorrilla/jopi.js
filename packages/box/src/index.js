import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
