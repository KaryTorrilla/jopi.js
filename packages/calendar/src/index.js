import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
