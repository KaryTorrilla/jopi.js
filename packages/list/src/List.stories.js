import React from 'react'
import { action } from '@storybook/addon-actions'

import { List, useFilterData } from '.'

export default {
  component: List,
  title: 'List',
}

const data = [
  {
    id: 22,
    content: {
      name: 'Agustin Iglesias',
      count: 10,
    },
  },
  {
    id: 23,
    content: {
      name: 'Josefina Golubicki',
      count: 43,
    },
  },
  {
    id: 8,
    content: {
      name: 'Wally',
      count: 45,
    },
  },
]

export const normal = () => (
  <List width={1 / 3}>
    <List.Item>Hola</List.Item>
    <List.Item>Good</List.Item>
    <List.Item>Bye</List.Item>
  </List>
)
export const selectable = () => (
  <List width={1 / 3}>
    <List.Item
      onClick={action('Data selected change')}
      sx={{ cursor: 'pointer' }}
    >
      Hola
    </List.Item>
    <List.Item
      onClick={action('Data selected change')}
      sx={{ cursor: 'pointer' }}
    >
      Good
    </List.Item>
    <List.Item
      onClick={action('Data selected change')}
      sx={{ cursor: 'pointer' }}
    >
      Bye
    </List.Item>
  </List>
)

export const searchable = () =>
  React.createElement(() => {
    const [filteredData, setValue] = useFilterData(data, 'content')

    return (
      <List width={1 / 3}>
        <List.Search onChange={(e) => setValue(e.target.value)} />
        {filteredData.map((user) => (
          <List.Item key={user.id}>{user.content.name}</List.Item>
        ))}
      </List>
    )
  })
