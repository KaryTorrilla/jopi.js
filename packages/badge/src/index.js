import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
