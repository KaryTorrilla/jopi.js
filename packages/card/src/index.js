import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
