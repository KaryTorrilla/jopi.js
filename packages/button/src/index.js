import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
