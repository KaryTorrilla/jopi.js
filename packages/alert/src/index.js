import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
