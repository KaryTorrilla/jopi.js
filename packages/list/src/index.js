import React, { forwardRef } from 'react'
import { Box, Flex } from '@oneloop/box'
import { Input } from '@oneloop/input'
import { Checkbox } from '@oneloop/checkbox'
import { useToggle, useFilterData } from '@oneloop/hooks'
import { Button, ButtonHoldPress, ButtonIcon } from '@oneloop/button'

const List = forwardRef(({ children, ...props }, ref) => (
  <Box
    ref={ref}
    as="ul"
    {...props}
    __css={{
      listStyleType: 'none',
      p: 0,
    }}
  >
    {children}
  </Box>
))

const ListInput = (props) => (
  <Flex as="span"
    sx={{
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      padding: '10px 6px 12px',
      gap: '8px',
    }}
  >
    <Input
      prefix={
        <span className='icon-buscar' style={{
          color: 'neutralGray3',
        }}/>
      }
      width={1}
      {...props}
      sx={{
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'flex-start',
        gap: '4px',
        backgroundColor: 'white',
        borderRadius: '8px',
        border: '1px solid #C9D1D5',
        height: '28px',
        color: 'neutralGray3',
        fontFamily: 'Nunito Sans',
        fontWeight: 400,
        fontSize: '12px',
        lineHeight: '16px',
      }}
    />
  </Flex>
)

const ListItem = ({ children, hover = true, ...props }) => (
  <Box
    as="li"
    width={1}
    {...props}
    __css={{
      py: '10px',
      px: '16px',
      display: 'inline-flex',
      // justifyContent: 'start',
      alignItems: 'center',
      fontFamily: 'primary',
      // ':hover': hover && { bg: 'neutral.1' },
    }}
  >
    <Checkbox />
    {children}
  </Box>
)

const ListDefault = ({ children, disabled = false, hover = true, variantSize = 'dropdownSizeNormal', ...props }) => {
  return (
    <Box
      as="li"
      width={1}
      {...props}
      __css={{
        display: 'inline-flex',
        justifyContent: 'start',
        alignItems: 'center',
      }}
    >
      <div style={{ width: '100%' }}>
        <Button variant={[disabled ? 'mainItemSmallListDisabled' : 'mainItemSmallList', variantSize]} >{children}</Button>
      </div>
    </Box>
  )
}

const ListMultiselect = ({ children, disabled = false, hover = true, isActive, variantSize = 'dropdownSizeNormal', ...props }) => {
  const [active, toggle] = useToggle(false)
  if (isActive && !active) {
    toggle(true)
  }

  return (
    <Box
      as="li"
      width={1}
      {...props}
      __css={{
        display: 'inline-flex',
        justifyContent: 'start',
        alignItems: 'center',
      }}
    >
      <div style={{ width: '100%' }}>
        <ButtonHoldPress variant={[disabled ? 'mainItemSmallListDisabled' : 'mainItemSmallList', variantSize]} active={active} onClick={toggle} text={children} hasCheckbox disabled={disabled}/>
      </div>
    </Box>
  )
}

const ListIcon = ({ children, disabled = false, hover = true, isActive, icon, variantSize = 'dropdownSizeNormal', ...props }) => {
  return (
    <Box
      as="li"
      width={1}
      {...props}
      __css={{
        display: 'inline-flex',
        justifyContent: 'start',
        alignItems: 'center',
      }}
    >
      <div style={{ width: '100%' }}>
        <ButtonIcon variant={[disabled ? 'mainItemSmallListDisabled' : 'mainItemSmallList', variantSize]} icon={icon} text={children}/>
      </div>
    </Box>
  )
}

List.Default = ListDefault
List.Multiselect = ListMultiselect
List.Icon = ListIcon
List.Search = ListInput
List.Item = ListItem
List.displayName = 'List'

export { List, useFilterData }
