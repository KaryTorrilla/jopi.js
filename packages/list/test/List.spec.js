import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

import { List } from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('List', () => {
  beforeAll(() => {
    const div = document.createElement('div')
    window.domNode = div
    document.body.appendChild(div)
  })

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <List>
          <List.Search />
          <List.Item>One</List.Item>
          <List.Item>Two</List.Item>
          <List.Item>Three</List.Item>
        </List>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ListDefault', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <List width={1 / 3}>
          <List.Default disabled>Item 1</List.Default>
          <List.Default>Item 2</List.Default>
          <List.Default>Item 3</List.Default>
        </List>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ListMultiselect', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <List width={1 / 3}>
          <List.Multiselect disabled>Item 1</List.Multiselect>
          <List.Multiselect>Item 2</List.Multiselect>
          <List.Multiselect>Item 3</List.Multiselect>
        </List>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ListMultiselectClick', () => {
  test('renders correctly', () => {
    const wrapper = mount(
      <List width={1 / 3}>
        <List.Multiselect disabled>Item 1</List.Multiselect>
        <List.Multiselect id="itemToClick" isActive>
          Item 2
        </List.Multiselect>
        <List.Multiselect>Item 3</List.Multiselect>
      </List>,
      {
        attachTo: window.domNode,
      }
    )
    expect(wrapper).toMatchSnapshot()
    const itemToClick = document.getElementById('itemToClick')
    itemToClick.click()
  })
})

describe('ListIcon', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <List width={1 / 3}>
          <List.Icon icon="icon-agente" disabled>
            Item 1
          </List.Icon>
          <List.Icon icon="icon-agente">Item 2</List.Icon>
          <List.Icon icon="icon-agente">Item 3</List.Icon>
        </List>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
