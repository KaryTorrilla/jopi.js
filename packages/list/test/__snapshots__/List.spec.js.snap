// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  list-style-type: none;
  padding: 0;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 14px;
  padding-bottom: 14px;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 10px;
  border: none;
  box-shadow: none;
  font-family: primary;
  display: block;
  font-size: 14px;
  height: 36px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 11px;
  padding-bottom: 11px;
  outline: none;
  background-color: inputBackground;
  border-radius: circle;
  background-color: rgba(0,0,0,0.04);
  margin-bottom: 0;
  width: 100%;
}

.c5::-webkit-input-placeholder {
  color: inputPlaceholderColor;
}

.c5::-moz-placeholder {
  color: inputPlaceholderColor;
}

.c5:-ms-input-placeholder {
  color: inputPlaceholderColor;
}

.c5::placeholder {
  color: inputPlaceholderColor;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-left: 16px;
  padding-right: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: primary;
  width: 100%;
}

.c7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  width: 18px;
  height: 18px;
}

.c8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  overflow: hidden;
  width: 100%;
  height: 100%;
}

.c12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 2px;
  color: gray;
}

input:checked ~ .c0 {
  color: primary;
}

input:focus ~ .c0 {
  box-shadow: active;
}

.c9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 2px;
  color: gray;
  width: 16px;
  height: 16px;
}

input:checked ~ .c0 {
  color: primary;
}

input:focus ~ .c0 {
  box-shadow: active;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  display: none;
}

input:checked:not(:disabled) ~ .c10 {
  display: block;
}

.c14 {
  display: none;
}

input:not(:checked):not(:disabled) ~ .c13 {
  display: block;
}

.c18 {
  display: none;
}

input:checked:disabled ~ .c17 {
  display: block;
}

.c16 {
  display: none;
}

input:disabled:not(:checked) ~ .c15 {
  display: block;
}

<ul
  className="c0 c1"
>
  <span
    className="c0 c2 c3"
  >
    <div
      className="c0 c4"
    >
      <input
        className="c0 c5"
        disabled={false}
        type="text"
        width={1}
      />
    </div>
  </span>
  <li
    className="c0 c6"
    width={1}
  >
    <div
      className="c0 c7"
    >
      <input
        className="c0 c8"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        className="c0 c9 c10 c11 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#6A88F2"
          height="15"
          rx="4"
          ry="4"
          stroke="#6A88F2"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="white"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c13 c14 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="white"
          height="15"
          rx="4"
          ry="4"
          stroke="#C1CCD0"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c15 c16 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c17 c18 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="#C1CCD0"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
    </div>
    One
  </li>
  <li
    className="c0 c6"
    width={1}
  >
    <div
      className="c0 c7"
    >
      <input
        className="c0 c8"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        className="c0 c9 c10 c11 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#6A88F2"
          height="15"
          rx="4"
          ry="4"
          stroke="#6A88F2"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="white"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c13 c14 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="white"
          height="15"
          rx="4"
          ry="4"
          stroke="#C1CCD0"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c15 c16 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c17 c18 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="#C1CCD0"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
    </div>
    Two
  </li>
  <li
    className="c0 c6"
    width={1}
  >
    <div
      className="c0 c7"
    >
      <input
        className="c0 c8"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        className="c0 c9 c10 c11 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#6A88F2"
          height="15"
          rx="4"
          ry="4"
          stroke="#6A88F2"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="white"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c13 c14 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="white"
          height="15"
          rx="4"
          ry="4"
          stroke="#C1CCD0"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c15 c16 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
      </svg>
      <svg
        aria-hidden="true"
        className="c0 c9 c17 c18 c0 c12"
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#EFF2F3"
          height="15"
          rx="4"
          ry="4"
          stroke="#EFF2F3"
          width="15"
          x="0.5"
          y="0.5"
        />
        <path
          d="M 5,8 L 7,10 L 12,5"
          stroke="#C1CCD0"
          strokeLinecap="round"
          strokeWidth="1.5"
        />
      </svg>
    </div>
    Three
  </li>
</ul>
`;
